js/dcf-tabs.js

```diff
@@ -0,0 +1,112 @@
+        // Add class to each tab
+        tab.classList.add('dcf-tab', 'dcf-d-block');
+        // Prefix each tab within its parent tab group with the corresponding uuid.
+        let nextTab = tabIndex + DCFUtility.magicNumbers('int1');
+        tab.setAttribute('id', uuid.concat('-tab-', nextTab));
+        // Add role to each tab
+        tab.setAttribute('role', 'tab');
+        // Add tabindex to each tab
+        tab.setAttribute('tabindex', '-1');
+        // Add class to each tab's parent (list item)
+        tab.parentNode.classList.add('dcf-tabs-list-item', 'dcf-mb-0');
+        // Add role to each tab's parent (list item)
+        tab.parentNode.setAttribute('role', 'presentation');
+        // Handle clicking of tabs for mouse users
+        tab.addEventListener('click', (clickEvent) => {
+          clickEvent.preventDefault();
+          let currentTab = tabList.querySelector('[aria-selected]');
+          if (clickEvent.currentTarget !== currentTab) {
+            this.switchTab(currentTab, clickEvent.currentTarget, tabs, panels);
+          }
+        });
+        // Handle keydown events for keyboard users
+        tab.addEventListener('keydown', (keydownEvent) => {
+          // Get the index of the current tab in the tabs node list
+          let index = Array.prototype.indexOf.call(tabs, keydownEvent.currentTarget);
+          // Work out which key the user is pressing and
+          // Calculate the new tab's index where appropriate
+          let dir = 0;
+          if (keydownEvent.which === DCFUtility.magicNumbers('arrowLeftCode')) {
+            dir = index - DCFUtility.magicNumbers('int1');
+          } else if (keydownEvent.which === DCFUtility.magicNumbers('arrowRightCode')) {
+            dir = index + DCFUtility.magicNumbers('int 1');
+          } else if (keydownEvent.which === DCFUtility.magicNumbers('arrowDownCode')) {
+            dir = 'down';
+          } else {
+            dir = null;
+          }
+
+
+          if (dir !== null) {
+            keydownEvent.preventDefault();
+            // If the down key is pressed, move focus to the open panel,
+            // otherwise switch to the adjacent tab
+            if (dir === 'down') {
+              panels[tabIndex].focus();
+            } else if (tabs[dir]) {
+              this.switchTab(keydownEvent.currentTarget, tabs[dir], tabs, panels);
+            }
+          }
+        });
+      });
+
+      // Add tab panel semantics and hide them all in each tab group.
+      Array.prototype.forEach.call(panels, (panel, panelIndex) => {
+        // Set role to each tab panel
+        panel.setAttribute('role', 'tabpanel');
+        // Set tabindex to let panel be focused
+        panel.setAttribute('tabindex', '-1');
+        // Add class to each tab panel
+        panel.classList.add('dcf-tabs-panel');
+        // Declare which tab labels each panel
+        panel.setAttribute('aria-labelledby', tabs[panelIndex].id);
+        panel.classList.add('dcf-mb-5');
```

Comment:
While each tab panel certainly needs some spacing around it, I think we should leave that for the theme or individual developer to style. I'd suggest removing this line.